### PR TITLE
[4.0] Make mediamanager usable on mobile

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue #27975

### Summary of Changes
Attempting to create working mobile view.

### Testing Instructions
go to administrator/index.php?option=com_media&path=local-0:/
check desktop & mobile view.
apply pr
run `npm run build:com_media`
recheck desktop & mobile views in both portrait & landscape.

Known issues:

- [x] **Currently sidebar doesn't resize as you drill down deeper in desktop view**

- [x] breadcrumb always present.

- [x] breadcrumbs don't wrap properly

- [x] info flyout too narrow in mobile view

- [ ] On some mobile screen file path doesn't wrap.

- [ ] zoom in/out doesn't work properly.  ( sometimes + = out )

### Actual result BEFORE applying this Pull Request
note8

### Expected result AFTER applying this Pull Request
image
note8
image

### CODE REVIEW

**Comment by PhocaCz** [comment]
Hi, before patch:
img

after patch:
img

**Comment by N6REJ** [comment]
@PhocaCz this was NOT on a mobile device right?  It was simulated?

**Comment by N6REJ** [comment]
Got it to here so far 
note8

**Comment by brianteeman** [comment]
Can you explain what you mean by this please
>  Currently sidebar doesn't resize as you drill down deeper

**Comment by PhocaCz** [comment]
> @PhocaCz this was NOT on a mobile device right? It was simulated?

Yes, simulated

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <template>
-  <div>
-    <div
+  <div
```
Comment (richard67) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue]: Is it by purpose that the outer div is removed here?

If so, it needs to indent everything which was inside this div by 1 level less, i.e. 2 spaces less.

But removing this div could be the reason why system tests for this PR consistently fail when trying to toggle the info bar in the media manager.

Have restarted drone again to confirm.

@N6REJ Please check and report back if the div really has to be removed.

File: administrator/components/com_media/resources/styles/components/_layout.scss [hunk 2]
```diff
@@ -4,6 +4,7 @@
   flex-wrap: wrap;
   margin-right: -($col-gutter-width / 2);
   margin-left: -($col-gutter-width / 2);
+
```
Comment (richard67) [inline administrator/components/com_media/resources/styles/components/_layout.scss:8]:
```suggestion
```
Revert adding this empty line before the closing curly bracket.

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue (hunk 1 above)
Comment (richard67) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: Again drone fails at the same step:



`I wait for element visible {"class":"media-infobar"}` ... waiting for Godot?

Comment (richard67) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: Maybe the info bar just has the wrong initial state? The test says with the first `I open infobar`: `I see element {"class":"media-infobar"}`. Then it says with the second `I open infobar`: `I click {"class":"media-toolbar-info"}`, and then `I wait for element visible {"class":"media-infobar"}`. So maybe it needs to be closed initially?

Comment (richard67) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: @N6REJ I've just done a real test, went to media manager with desktop screen site, selected an image and used the info "I". It opens something but that's empty, i.e. results in white space. So this PR here definitely breaks it.

Comment (richard67) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: Either you have to add back the outer div again, or it needs to modify fome js, I would guess, to fill in the info into the right div or whatever else element.

Comment (N6REJ) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: @richard67 ok ty I'll check the code more.
@brianteeman when you expand the sidebar by drilling down it doesn't expand.

Comment (richard67) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: @N6REJ Not only check the code. Do a real test. And real test doesn't mean just to size down the screen size and see if it looks nice.

Real test means check also in desktop size if things work like before. E.g. there should be an area where you can do drag and drop to for uploading images. Does that still work? Does the info button work? the system tests which are failing for this PR do nothing else, the simulate clicks on buttons and check the result. so if you wanna see why a system test goes wrong and the drone logs and screenshots don't show the reason, then make a new install and "manually" do what the system tests tell that they do, e.g. "Test that it is possible to toggle the info bar".

Such real tests are not to be done by the PR testers. They are a requirement for submitting a PR. When you submit a PR, it is expected that you have tested it, and this includes also having tested that it does not break functionality.

**Comment by brianteeman** [comment]
Try to move media-breadcrumbs onto a second row for mobile views and probably get rid of the search on mobile views as well

**Comment by jiweigert** [comment]
hiding the search input field on mobile view and replace it with a floating search-icon instead would be a good idea.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31140">issues.joomla.org/tracker/joomla-cms/31140</a>.</sub>

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <template>
-  <div>
-    <div
+<div>
+  <div
```
Comment (richard67) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue]:
```suggestion
  <div>
    <div
```

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 4]
```diff
@@ -84,7 +84,7 @@
       </div>
     </div>
     <media-infobar ref="infobar" />
-  </div>
+</div>
```
Comment (richard67) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue]:
```suggestion
  </div>
```

**Comment by N6REJ** [comment]
> hiding the search input field on mobile view and replace it with a floating search-icon instead would be a good idea.
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/31140](https://issues.joomla.org/tracker/joomla-cms/31140).

idk how to replace it.. hiding it is easy part

**Comment by N6REJ** [comment]
> Try to move media-breadcrumbs onto a second row for mobile views and probably get rid of the search on mobile views as well

do we even need/wants breadcrumbs on desktop?

**Comment by N6REJ** [comment]
as of https://github.com/joomla/joomla-cms/pull/31140/commits/110e506704ba6c2cc6bb6aec0d325b8d8d3f8199
Screen Shot 2020-10-19 at 17 51 50
image

**Comment by N6REJ** [comment]
based on feedback search removed on < md
image

**Comment by N6REJ** [comment]
paused here for the night 
image

**Comment by N6REJ** [comment]
Can someone check RTL for breadcrumb.scss.  Made several changes in that file.
idk how to fix the wordwrap for file path since it would have to break on / and it seems to be device specific.
lastly, zoom in/out is not something "I" can fix, so that will have to be for someone else in another pr or supply the code here.
All in all I"m going to say this is as complete as can be for this pr.
READY FOR TESTING

**Comment by N6REJ** [comment]
@richard67 I don't understand what drone is trying to say.  Perhaps class in tester needs to change?

**Comment by richard67** [comment]
@N6REJ I think I had explained you how to find the drone logs.

The system test fails here at a step labelled "Test that it is possible to navigate to a subfolder using breadcrumb.".

So I guess that is related to your changes for the breadcrumbs.

**Comment by N6REJ** [comment]
@richard67 yep, test is going to need to be modified in some way.  The only time breadcrumbs display now is in mobile view.
```
	/**
	 * Test that it's possible to navigate to a subfolder using breadcrumb.
	 *
	 * @param   Media  $I
	 *
	 * @since   4.0.0
	 *
	 * @throws Exception
	 */
	public function navigateUsingBreadcrumb(Media $I)
	{
		$I->wantToTest('that it is possible to navigate to a subfolder using breadcrumb.');
		$I->amOnPage(MediaListPage::$url . 'banners');
		$I->waitForMediaLoaded();
		$I->clickOnLinkInBreadcrumb('images');
		$I->waitForMediaLoaded();
		$I->seeInCurrentUrl(MediaListPage::$url);
		$I->seeContents($this->contents['root']);
	}
```

File: administrator/components/com_media/resources/scripts/components/breadcrumb/breadcrumb.vue [hunk 5]
```diff
@@ -4,11 +4,13 @@
     role="navigation"
     :aria-label="translate('COM_MEDIA_BREADCRUMB_LABEL')"
   >
-    <ol>
+    <ol
+      class="breadcrumb">
```
Comment (Quy) [inline administrator/components/com_media/resources/scripts/components/breadcrumb/breadcrumb.vue]:
```suggestion
    <ol class="breadcrumb">
```

File: administrator/components/com_media/resources/styles/components/_layout.scss [hunk 6]
```diff
@@ -34,11 +36,18 @@
...
   height: calc(100vh - 160px);
+  margin-left: 1rem;
   overflow: hidden;
   background-color: #fff;
   border: 1px solid $border-color;
   border-radius: $border-radius;
   box-shadow: $col-box-shadow;
+
+  /* Small devices (Portrait and Landscape, < 576px) */
+  @media (max-width: 576px) {
```
Comment (Quy) [inline administrator/components/com_media/resources/styles/components/_layout.scss]: Use variable. See line 29.

**Comment by brianteeman** [comment]
Are there updates still outstanding or is that label resolved and should be removed?

**Comment by richard67** [comment]
Last changes after @Quy has added the "Updates Requested" label were only clean branch updates, so I am not sure.

@Quy Could you check and report back? Thanks in advance.

File: administrator/components/com_media/resources/styles/components/_layout.scss (hunk 6 above)
Comment (Quy) [reply administrator/components/com_media/resources/styles/components/_layout.scss]: @richard67 This was marked resolved without action.

Comment (richard67) [reply administrator/components/com_media/resources/styles/components/_layout.scss]: Unresolved. @N6REJ Please check.

**Comment by Quy** [comment]
Also, there are 2 outstanding items so I assume it is not ready for testing.

File: administrator/components/com_media/resources/styles/components/_layout.scss (hunk 6 above)
Comment (N6REJ) [reply administrator/components/com_media/resources/styles/components/_layout.scss]: fixed https://github.com/joomla/joomla-cms/pull/31140/commits/b5c53635b739b12d3af0788b330c27036fb26fcc

**Comment by N6REJ** [comment]
> Also, there are 2 outstanding items so I assume it is not ready for testing.

I don't see 2, am I missing something?

**Comment by Quy** [comment]
Under known issues, are these to be fixed in this PR?

> On some mobile screen file path doesn't wrap.

> zoom in/out doesn't work properly. ( sometimes + = out )

**Comment by N6REJ** [comment]
@Quy no.  I honestly don't know vue even remotely enough to tackle them.  I wish I did.
I've tackled the path as best as I know how. it's only in the info window
image

**Comment by ceford** [comment]
Problems I see today (2020-11-18):
On 1440 px wide screen: Magnify buttons give wrong sequences; I button opens slider to cover about 75% of div width including image icons.
On simulated mobile: improving but still needs work. Keep going!
screen shot 2020-11-18 at 19 05 06
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31140">issues.joomla.org/tracker/joomla-cms/31140</a>.</sub>

**Pull request closed without merging**